### DISCUSSION TITLE
Inconsistent framebuf format

### DISCUSSION THREAD

**Discussion by AncientJames** [micropython/micropython#9388] [Libraries & Drivers]
I'm not sure if this is the place to flag it, but the framebuf FRAMEBUF_GS2_HMSB format is arranged with msb to the right, the reverse of the other formats.

**Comment by peterhinch** [comment]
Are you sure? Looking at [the sourcecode](https://github.com/micropython/micropython/blob/30e50ab1951680a046c224c12f7cc29aed410e2a/extmod/modframebuf.c#L150) it's hard to see how this can be the case.

**Reply by AncientJames** [reply]
Sorry, it's GS4 that's the wrong way round.

**Comment by peterhinch** [comment]
I suggest you back up your assertion by pointing to an error in the very simple [getpixel](https://github.com/micropython/micropython/blob/30e50ab1951680a046c224c12f7cc29aed410e2a/extmod/modframebuf.c#L176) source. As far as I can see there is no possible mechanism for bit reversal.

**Reply by AncientJames** [reply]
GS2 shifts right with increasing x. GS4 tests bit 0 and then returns the right shifted result if it's unset - ie with decreasing x.

This shows the behaviour:
```
import framebuf

def fbtest(fmt):
    ba = bytearray(8)
    fb = framebuf.FrameBuffer(ba, 8, 1, fmt)
    fb.fill(0)
    fb.pixel(0, 0, 255)
    return ba[0]

print(f'{fbtest(framebuf.MONO_HLSB):08b}  MONO_HLSB')
print(f'{fbtest(framebuf.MONO_HMSB):08b}  MONO_HMSB')
print(f'{fbtest(framebuf.GS2_HMSB):08b}  GS2_HMSB')
print(f'{fbtest(framebuf.GS4_HMSB):08b}  GS4_HMSB')
```

Resulting in:
```
10000000  MONO_HLSB
00000001  MONO_HMSB
00000011  GS2_HMSB
11110000  GS4_HMSB
```

**Comment by peterhinch** [comment]
OK, now I understand. 

I think we need new constants with a different naming convention and much better documentation. We can't delete the existing constants because it would break code. I suggest `GS2_RJ == GS2_HMSB` and `GS4_LJ == GS4_HMSB` where LJ and RJ mean left and right justification for multi-bit colors.

Do you want to raise a PR or shall I?

Any other thoughts on naming?
